src/Lumi/Components/Select/SelectBackend.purs

```diff
@@ -256,23 +303,22 @@ selectBackend = make component
                       when self.state.isOpen do
                         eventTargetIsChild <- isOrContainsNode rootRef target
                         when (not eventTargetIsChild) do
-                          childProps.closeSelect
+                            send self CloseSelect
+                            send self Blur
               }
             , { eventType: EventType "keydown"
               , options: { capture: false, once: false, passive: false }
               , handler: \e -> do
-                  if not self.state.isOpen
-                    then pure unit
-                    else keydownEventHandler e
+                  when self.state.isActive do
+                    keydownEventHandler e
```

Comment:
This plus the updated key handlers below allow the dropdown to be re-opened after a selection is made using the up/down arrow keys or ctrl+p/n (my own keyboard habits). Tab/shift-tab also work, but only if the dropdown is already open to allow normal form field tabbing.